### ISSUE TITLE
[5.6.8] Not able to use tightenco/collect

### ISSUE THREAD

**Issue by mpociot** [laravel/framework#23420]
- Laravel Version: 5.6.8

### Description:
Because we now treat `tightenco/collect` as a conflict (PR https://github.com/laravel/framework/pull/23379) composer installs both packages and adds them to the autoloader.

The problem is that the autoloader sequence is wrong and puts `tightenco/collect` in front, therefore loading the `tightenco/collect` helper methods instead of Laravel's helpers.

```php
'fe62ba7e10580d903cc46d808b5961a4' => __DIR__ . '/..' . '/tightenco/collect/src/Collect/Support/helpers.php',
'caf31cc6ec7cf2241cb6f12c226c3846' => __DIR__ . '/..' . '/tightenco/collect/src/Collect/Support/alias.php',
'f0906e6318348a765ffb6eb24e0d0938' => __DIR__ . '/..' . '/laravel/framework/src/Illuminate/Foundation/helpers.php',
'58571171fd5812e6e447dce228f52f4d' => __DIR__ . '/..' . '/laravel/framework/src/Illuminate/Support/helpers.php',
````

Because of this, the wrong collection classes will be passed internally:

```
Type error: Argument 1 passed to Illuminate\Routing\Router::sortMiddleware() must be an instance of Illuminate\Support\Collection, instance of Tightenco\Collect\Support\Collection given, called in /Users/marcel/Code/chatwidget/vendor/laravel/framework/src/Illuminate/Routing/Router.php on line 676
```

### Steps To Reproduce:

```bash
laravel new testapp
cd testapp
composer require spatie/crawler
open http://testapp.test
```

See error:

```
Type error: Argument 1 passed to Illuminate\Routing\Router::sortMiddleware() must be an instance of Illuminate\Support\Collection, instance of Tightenco\Collect\Support\Collection given, called in /Users/marcel/Code/chatwidget/vendor/laravel/framework/src/Illuminate/Routing/Router.php on line 676
```

**Comment by SjorsO** [comment]
Could someone help me understand why `tightenco/collect` is used at all in the frameworks composer.json?

**Comment by sisve** [comment]
We're declaring that we're incompatible with the older (<5.5.33) version of tightenco/collect since that version was using identical namespaces and class names as Laravel provided. It's not being pulled into your project by the root composer.json file, but the spatie/crawler project.

So, the tightenco/collect package declare helpers that are not compatible with Laravel. I believe the current stance is that this is someone else's fault. They shouldn't use our function names. Booo them! Booo! https://github.com/laravel/framework/issues/20370 https://github.com/laravel/internals/issues/709

On a more serious note, the only way around this is to namespace/prefix all our helpers with something relevant. We cannot control what third party code is declaring, but we can reduce the area of conflict.

Another problem is that we're conditionally defining the helper functions. This means that the code doesn't fail in helpers.php, but instead later when someone discovers that the function they called wasn't the one they intended to call.

**Comment by sisve** [comment]
The clarify, the problem is not that we're declaring a conflict in our composer.json file, that doesn't pull down the tightenco/collect package. You get it because you're installing spatie/crawler with depends on it.

The problem is that you've got a package that has conflicting helpers. That's not allowed. Boo!

**Comment by mattstauffer** [comment]
FWIW, we consider this an issue that collect and the collect team are are taking the responsibility to fix. If it requires a change at the framework level we will attempt to PR it but I believe it won’t.

**Comment by rykersouthwick** [comment]
I installed a captcha package for Laravel and got the exact same problem.

**Comment by repat** [comment]
@mattstauffer any updates/ETA on this, we'd really like to use spaties mixed-content-scanner ;-)

**Comment by tomcoonen** [comment]
@repat Please see https://github.com/tightenco/collect/issues/92 , a fix is released.

**Comment by tomcoonen** [comment]
@mpociot this issue can be closed i guess :)

**Comment by mpociot** [comment]
Yeah - this is now fixed in 5.6.12 of tightenco/collect - https://github.com/tightenco/collect/releases/tag/v5.6.12

**Comment by loslosbaby** [comment]
Hiya, somewhat of a modern PHP noob here.  What's the bottom-line fix for this?  I did a composer update and now my valet is broken: 

'''
Parse error: parse error in /Users/loslosbaby/.composer/vendor/tightenco/collect/src/Illuminate/Support/Collection.php on line 154
'''
Etc.  Do I understand that I need to look for this in one of the composer files and remove it?
